domain/src/main/java/edu/nextstep/camp/caculator/domain/Expression.kt

```diff
@@ -1,29 +1,95 @@
 package edu.nextstep.camp.caculator.domain
 
-class Expression (
-    private val expressionString: String) {
-    private val expressionList = mutableListOf<String>()
+import java.lang.StringBuilder
+import java.util.*
 
-    init {
-        setExpression(this.expressionString)
+class Expression {
+    private var expressionString: StringBuilder = StringBuilder()
+
+    fun getExpressionString(): String = expressionString.toString()
+
+    fun toExpressionData(): ExpressionData {
+        val queue: Queue<String> = LinkedList()
+        queue.addAll(expressionString.split(EXPRESSION_SEPARATOR))
+
+        return generateExpressionData(queue)
+    }
+
+    fun addExpression(expression: String) {
+        require(expression.isNotEmpty()
+                && (isInteger(expression)
+                || AVAILABLE_INPUT_OPERATORS.contains(expression)))
+        {"Expression is empty or wrong expression"}
+
+        when {
+            isInteger(expression) -> addValueExpression(expression)
+            AVAILABLE_INPUT_OPERATORS.contains(expression) -> addValueOperator(expression)
+        }
+    }
+
+    fun isValidExpression(): Boolean = AVAILABLE_INPUT_OPERATORS.contains(expressionString.last()).not()
+
+    fun clear() {
+        expressionString.clear()
+    }
+
+    fun deleteLastExpression() {
+        with(this.expressionString) {
+            when {
+                isEmpty() -> return
+                length < 2 -> clear()
+                get(length-2) == EXPRESSION_SEPARATOR -> deleteRange(length-2, length)
+                else -> deleteRange(length-1, length)
+            }   
+        }
+    }
+
+    private fun isInteger(s: String): Boolean {
+        return try {
+            s.toInt()
+            true
+        } catch (e: NumberFormatException) {
+            false
+        }
+    }
```

Comment:
```suggestion
    private fun isInteger(s: String )= try {
        s.toInt()
        true
    } catch (e: NumberFormatException) {
        false
    }
```

함수 내용이 return 하나인 표현식이라면 좀더 간결하게 = 를 활용해 보세요!